src/LearnMe.Web/Startup.cs

```diff
@@ -38,17 +41,23 @@ public void ConfigureServices(IServiceCollection services)
 
 
             services.AddControllersWithViews();
-            // In production, the Angular files will be served from this directory
             services.AddSpaStaticFiles(configuration =>
             {
                 configuration.RootPath = "ClientApp/dist";
             });
 
             services.AddDbContext<ApplicationDbContext>(options => options.UseSqlServer(Configuration.GetConnectionString("LearnMeDatabase"), b=> b.MigrationsAssembly("LearnMe.Web")));
 
-            services.AddScoped<IGoogleAPIconnection, GoogleAPIconnection>();
+            services.AddSingleton<ITokenService, TokenService>();
+            services.AddSingleton<IToken>(provider =>
+            {
+                var tokenService = provider.GetService<ITokenService>();
+                var token = tokenService.GetToken().GetAwaiter().GetResult();
```

Comment:
1. to nie może być singletonem bo dla 10 tysięcy użytkowników będzie jedna instancja `IToken`. a z tego co rozumiem, token ma być per użytkownik (`UserCredential`) a nie per instancja apki
2. To jest totalnie złe. Startup uruchamia się tylko raz i będziesz i tak miała tylko jeden `IToken` mimo tego, że dasz `Scoped` czy inny modyfikator na ten serwis. Doczytaj jak działa .NET Core, czym jest klasa Program, klasa Startup,  co robi metoda ConfigureServices i metoda Configure w Startupie. 

Ta linijka z wcyiąganiem tokena powinna istnieć w middleware. Już ci to pisałem na priv. Middleware może byc asynchroniczny.

https://docs.microsoft.com/pl-pl/aspnet/core/fundamentals/middleware/?view=aspnetcore-3.1

Reply:
![image](https://user-images.githubusercontent.com/37336963/95056944-bb0ede00-06f5-11eb-8360-57ad7c36a49e.png)